docs/gitworkflow.rst

```diff
@@ -22,25 +22,30 @@ developers are:
 
 * In order to ease integration, feature branches should be **short living** and
   pull requests should only contain a **reasonably small amount** of
-  changes. Try to chop your implementation into small self containing changes and
-  issue those in separate consecutive pull requests. (No one likes or is
+  changes. Try to chop your implementation into small self containing changes
+  and issue those in separate consecutive pull requests. (No one likes or is
   efficiently able to review a code change containing thousands of lines.)
 
 
 In order to add a feature, you have to do the following steps:
 
-#. Fork the official (upstream) repository and set up your own repository. (This
-   step you have to do only once.)
+#. Fork the official (upstream) repository and clone it. (This step you have to
+   do only once.)
 
-#. Make sure, your `master` branch is synchronised to the upstream `master`.
+#. Synchronize your `master` branch (``origin/master``) to be identical to
+   upstream `master` (``upstream master``).
 
 #. Derive a feature branch from the `master` branch of your forked project.
 
-#. Develop your feature in your feature branch.
+#. Develop and finish your feature in your feature branch.
 
-#. When your feature implementation has finished, make sure, it integrates well
-   into the most recent version of the code. (The code may have evolved while
-   you were implementing your feature.)
+#. Squash intermediate irrelevant commits of your feature branch, so that your
+   branch only contains 1-2 relevant ones.
+
+#. Synchronize your `master` branch (``origin/master``) to be identical to
+   upstream `master` (``upstream master``).
+
+#. Rebase your feature branch on the update master branch.
```

Comment:
Quite a change in the git policy. While I personally prefer rebasing over merging master, the rebasing workflow can be tricky for users new to git.

What is the reasoning for the change in the git development strategy?

Reply:
Second the comment, I think we should discuss this.